src/SDK/Language/Dart.php

```diff
@@ -251,6 +252,23 @@ public function getParamExample(array $param): string
         };
     }
 
+    public function getPermissionExample(string $example): string
+    {
+        $permissions = [];
+        foreach ($this->extractPermissionParts($example) as $permission) {
+            $args = [];
+            if ($permission['id'] !== null) {
+                $args[] = "'" . $permission['id'] . "'";
+            }
+            if ($permission['innerRole'] !== null) {
+                $args[] = "'" . $permission['innerRole'] . "'";
+            }
+            $argsString = implode(', ', $args);
+            $permissions[] = 'Permission.' . $permission['action'] . '(Role.' . $permission['role'] . '(' . $argsString . '))';
+        }
+        return '[' . implode(', ', $permissions) . ']';
+    }
```

Comment:
We can put this in the root language class, and add a couple of getters for each language, `getStaticAccessOperator` (e.g. `.` for JS,  `::` for PHP) and `getArrayOf(string)`, then it'll be much easier to maintain